imitation/scripts.py

```diff
@@ -80,51 +81,62 @@ def show_plot_disc():
         plt.legend()
         _savefig_timestamp("plot_fight_loss_disc", interactive)
 
-    gen_ep_reward = []
-    rand_ep_reward = []
+    gen_ep_reward = defaultdict(list)
+    rand_ep_reward = defaultdict(list)
+    exp_ep_reward = defaultdict(list)
 
-    def add_plot_gen():
+    def add_plot_gen(env, name):
         if n_gen_plot_episodes <= 0:
             return
 
         gen_policy = trainer.gen_policy
         rand_policy = util.make_blank_policy(env)
+        exp_policy = trainer.expert_policies[-1]
 
         gen_rew = util.rollout.total_reward(
-                gen_policy, env, n_episodes=n_gen_plot_episodes
-                ) / n_gen_plot_episodes
+            gen_policy, env, n_episodes=n_gen_plot_episodes
+        ) / n_gen_plot_episodes
         rand_rew = util.rollout.total_reward(
-                rand_policy, env, n_episodes=n_gen_plot_episodes
-                ) / n_gen_plot_episodes
-        gen_ep_reward.append(gen_rew)
-        rand_ep_reward.append(rand_rew)
+            rand_policy, env, n_episodes=n_gen_plot_episodes
+        ) / n_gen_plot_episodes
+        exp_rew = util.rollout.total_reward(
+            exp_policy, env, n_episodes=n_gen_plot_episodes
+        ) / n_gen_plot_episodes
+        gen_ep_reward[name].append(gen_rew)
+        rand_ep_reward[name].append(rand_rew)
+        exp_ep_reward[name].append(exp_rew)
         tf.logging.info("generator reward: {}".format(gen_rew))
         tf.logging.info("random reward: {}".format(rand_rew))
+        tf.logging.info("exp reward: {}".format(exp_rew))
 
     def show_plot_gen():
         if n_gen_plot_episodes <= 0:
             return
 
-        plt.title("Cartpole performance (expert=500)")
-        plt.xlabel("epochs")
-        plt.ylabel("Average reward per episode (n={})"
-                   .format(n_gen_plot_episodes))
-        plt.plot(gen_ep_reward, label="avg gen ep reward", c="red")
-        plt.plot(rand_ep_reward, label="avg random ep reward", c="black")
-        plt.legend()
-        _savefig_timestamp("plot_fight_epreward_gen", interactive)
+        for name in gen_ep_reward:
+            plt.title(name + " Performance")
+            plt.xlabel("epochs")
+            plt.ylabel("Average reward per episode (n={})"
+                       .format(n_gen_plot_episodes))
+            plt.plot(gen_ep_reward[name], label="avg gen ep reward", c="red")
+            plt.plot(rand_ep_reward[name],
+                     label="avg random ep reward", c="black")
+            plt.plot(exp_ep_reward[name], label="avg exp ep reward", c="blue")
+            plt.legend()
+            _savefig_timestamp("plot_fight_epreward_gen", interactive)
 
     add_plot_disc(False)
-    add_plot_gen()
+    add_plot_gen(env, "True Reward")
+    add_plot_gen(trainer.wrap_env_test_reward(env), "Learned Reward")
```

Comment:
Nice, I wonder what the learned reward plots look like :)

In the case of GAIL, the learned and train rewards are the same, right?